packages/models/files.js

```diff
@@ -7,6 +7,12 @@ const Files = new FilesCollection({
     debug: true,
     collectionName: 'files',
     protected(fileObj) {
+        var session = this.params.query.xmtok;
+        var userId  = (Meteor.server.sessions[session] && Meteor.server.sessions[session].userId) ? Meteor.server.sessions[session].userId : null;
```

Comment:
@thomasspiesser 

Users collection has not compound unique index for `services.resume.loginTokens.hashedToken`, this means your suggestion will have no positive effect. Possible side effects - slower response as this will involve two separate indexes.